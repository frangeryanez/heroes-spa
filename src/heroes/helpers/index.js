export * from './getHeroesByPublisher';
export * from './getHeroByName';