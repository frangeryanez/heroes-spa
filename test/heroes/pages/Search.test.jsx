import { fireEvent, render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Search } from '../../../src/heroes';

const mockedUseNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUseNavigate,
}));

describe('Test in <Search />', () => {
  beforeEach(() => jest.clearAllMocks() );

  test('Should be displayed correctly with default values', () => {
    const { container } = render(
      <MemoryRouter>
        <Search />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test('It should show Batman and the input with the value of the queryString', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman']}>
        <Search />
      </MemoryRouter>
    );
    
    const input = screen.queryByRole('textbox');
    expect(input.value).toBe('batman');
    
    const img = screen.getByRole('img');
    expect(img.src).toContain('/assets/heroes/dc-batman.jpg');

    const alert = screen.getByLabelText('alert-danger');
    expect(alert.style.display).toBe('none');
  });

  test('It should show an error if the hero is not found (batman123)', () => {
    render(
      <MemoryRouter initialEntries={['/search?q=batman123']}>
        <Search />
      </MemoryRouter>
    );

    const alert = screen.getByLabelText('alert-danger');
    expect(alert.style.display).toBe('');
  });

  test('Must call navigate to the new screen', () => {
    const inputValue = 'superman';

    render(
      <MemoryRouter initialEntries={['/search']}>
        <Search />
      </MemoryRouter>
    );

    const input = screen.getByRole('textbox');
    fireEvent.change(input, { 
      target: { name: 'searchText', value: inputValue }
    });
    
    const form = screen.getByRole('form');
    fireEvent.submit(form);
    
    expect(mockedUseNavigate).toHaveBeenCalledWith(`?q=${ inputValue }`)
  });
});