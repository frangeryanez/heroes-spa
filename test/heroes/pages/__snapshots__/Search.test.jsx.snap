// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <Search /> Should be displayed correctly with default values 1`] = `
<div>
  <h1>
    Search
  </h1>
  <hr />
  <div
    class="row"
  >
    <div
      class="col-5"
    >
      <h4>
        Searching
      </h4>
      <hr />
      <form
        aria-label="form"
      >
        <input
          autocomplete="off"
          class="form-control"
          name="searchText"
          placeholder="Search a hero"
          type="text"
          value=""
        />
        <div
          class="text-end"
        >
          <button
            class="text-center mt-2 btn btn-info"
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </div>
    <div
      class="col-7"
    >
      <h4>
        Results
      </h4>
      <hr />
      <div
        class="fade alert alert-primary animate__animated animate__fadeIn alert alert-primary show"
        role="alert"
      >
        Search a hero
      </div>
      <div
        aria-label="alert-danger"
        class="fade alert alert-primary animate__animated animate__fadeIn alert alert-danger show"
        role="alert"
        style="display: none;"
      >
        No hero with 
        <b />
      </div>
    </div>
  </div>
</div>
`;
